pulp_container/app/serializers.py

```diff
@@ -288,6 +293,12 @@ def validate(self, data):
                 )
         return validated_data
 
+    def is_base_path_valid(self, base_path):
+        """Check whether the passed repository base path is valid or not."""
+        component_re = re.compile(r"^[a-z0-9]+(?:(?:(?:[._]|__|[-]*)[a-z0-9]+)+)?$")
+        all_match = all(re.match(component_re, piece) for piece in base_path.split("/"))
+        return len(base_path) < 256 and all_match
```

Comment:
The `len` test seems to be cheap. Can we have it return before we start the regex operations? Also, i would raise the exceptions from this function instead of returning a bool (renaming it accordingly).
Actually, maybe follow this pattern: https://www.django-rest-framework.org/api-guide/serializers/#field-level-validation